app/Http/Controllers/UserController.php

```diff
@@ -373,11 +373,19 @@ public function postAdminEdit(Request $request)
 
         $user = User::find($user_id);
 
+        $oldRole = $user->role;
+
         // Set role for User
         $user->update([
-        'role' => $request->input('user_role'),
+            'role' => $request->input('user_role'),
         ]);
 
+        // If we are demoting from NetworkCoordinator, remove them from the list of coordinators for
+        // any networks they are currently coordinating.
+        if ($oldRole == Role::NETWORK_COORDINATOR && ($user->role == Role::HOST) || $user->role == Role::RESTARTER) {
```

Comment:
Is the operator precedence OK here?  Might be misreading it, but looks like anyone who is a Restarter will get detached from the network.

Reply:
It's wrong.  Thank you.  Fixed.